### PR TITLE
Complete get_dict_with_terminal_units_and_zones

### PR DESCRIPTION
**Pull request by weilixu**

### CODE REVIEW

**Review by yunjoonjung-PNNL — APPROVED** [review]
Looks great to me. Thank you for this effort!

File: docs/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/get_dict_with_terminal_units_and_zones.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
-**Description:** Returns a dictionary of zone IDs associated with each terminal unit in the RMR.   
+**Description:** Returns a dictionary of zone IDs associated with each terminal unit in the RMD.   
 
 **Inputs:**  
 - **RMR**: The RMR to evalute (B-RMR or P-RMR or U-RMR).    
 
 **Returns:**  
-- **get_dict_with_terminal_units_and_zones**: Returns a dictionary of zones associated with each terminal unit in the RMR, {terminal_unit_1.id: {"ZONE_LIST": [zone_1.id, zone_2.id, zone_3.id]}, terminal_unit_2.id: {"ZONE_LIST": [zone_4.id, zone_9.id, zone_30.id]}}
+- **get_dict_with_terminal_units_and_zones**: Returns a dictionary of zones associated with each terminal unit in the RMR, {terminal_unit_1.id: [zone_1.id, zone_2.id, zone_3.id], terminal_unit_2.id: [zone_4.id, zone_9.id, zone_30.id]}
```
Comment (charliepnnl) [inline docs/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/get_dict_with_terminal_units_and_zones.md:9]: I'm glad you flattened the dictionary.

File: rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/get_dict_with_terminal_units_and_zones.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    zone_1.id, zone_2.id, zone_3.id], terminal_unit_2.id: [zone_4.id, zone_9.id, zone_30.id]}
+
+    """
+    terminal_units_and_zones_dict = {}
+    for terminal in find_all("$..terminals[*]", rmi):
+        terminal_id = terminal["id"]
+        terminal_units_and_zones_dict[terminal_id] = find_all(
+            # search all zones that have terminal(s) match to this terminal id
+            # and return them in a list.
+            f'$..zones[*][?(@.terminals[*].id="{terminal_id}")].id',
```
Comment (charliepnnl) [inline rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/get_dict_with_terminal_units_and_zones.py:22]: Nicely done!
A slightly more Pythonic version would be:
```
return {
        terminal["id"]: find_all(
            # search all zones that have terminal(s) match to this terminal id
            # and return them in a list.
            f'$..zones[*][?(@.terminals[*].id="{terminal_id}")].id',
            rmi,
        )
        for terminal in find_all("$..terminals[*]", rmi)
    }
```
But I think you version is perfectly great too.

File: rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/test_get_dict_with_terminal_units_and_zones.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+                    "loop": "Purchased CHW Loop 1",
+                    "type": "CHILLED_WATER",
+                },
+            ],
+        }
+    ],
+}
+
+
+def test_get_dict_with_terminal_units_and_zones():
```
Comment (charliepnnl) [inline rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/test_get_dict_with_terminal_units_and_zones.py:226]: To match our test function naming convention, there should be another underscore after "test_".

**Review by charliepnnl — APPROVED** [review]

**Pull request merged**